### PR TITLE
Support subscribing to full dot

### PR DESCRIPTION
**Pull request by deltalejo**

### CODE REVIEW

File: examples/UbidotsSubscribeContext/UbidotsSubscribeContext.ino [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+char ssid[32 + sizeof("")] = "";
+char bssid[2*6 + 5 + sizeof("")] = "";
+char context[64 + sizeof("")] = "";
+
+/****************************************
+ * User Functions
+ ****************************************/
+
+void callback(char* topic, uint8_t* payload, unsigned int length) {
```
Comment (crisap94) [inline examples/UbidotsSubscribeContext/UbidotsSubscribeContext.ino:68]: To make it better to follow the code to the user, just declare the functions and the body of it define it after the loop function.

There will be less code to go through to get where the user would like to go. 

The Documentation should follow the standard https://developer.lsst.io/cpp/api-docs.html

File: examples/UbidotsSubscribeContext/UbidotsSubscribeContext.ino [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+   * payload as a serialized JSON object. Make sure you subscribe only to 
+   * complete dots, e.g. clientMQTT.ubidotsSubscribe("<device>", "<variable>", false),
+   * or make proper adjustments to support last value.
+   ****************************************************************************/
+  
+  Serial.println("--------------------------------");
+  
+  Serial.printlnf("Message from '%s': %.*s", topic, length, payload);
+  
+  char *jsonStr = (char *) malloc(length*sizeof(*payload) + sizeof(""));
```
Comment (crisap94) [inline examples/UbidotsSubscribeContext/UbidotsSubscribeContext.ino]: In our Library it is likely forbidden to use String objects, but the less we want in the examples files is to generate some kind of friction to the users when reading the code.

We do encourage whoever write these examples to write the code with the less traumatic functions. 

Long story short use String object in .ino files.

**Review by crisap94 — CHANGES_REQUESTED** [review]

File: examples/UbidotsSubscribeContext/UbidotsSubscribeContext.ino (hunk 1 above)
Comment (deltalejo) [reply examples/UbidotsSubscribeContext/UbidotsSubscribeContext.ino:68]: Ok!

File: examples/UbidotsSubscribeContext/UbidotsSubscribeContext.ino (hunk 2 above)
Comment (deltalejo) [reply examples/UbidotsSubscribeContext/UbidotsSubscribeContext.ino]: Hard to use String() in this case, will use strndup() to create a null-terminated string.

**Review by crisap94 — APPROVED** [review]

**Pull request merged**